src/connect/components/drive-view/drive-view.tsx

```diff
@@ -13,66 +13,54 @@ import {
 
 export type DriveType = 'public' | 'local' | 'cloud';
 
-export interface DriveTreeItem<T extends string = DefaultOptionId>
-    extends TreeItem<T> {
+export interface DriveTreeItem
+    extends TreeItem {
     type: ItemType.LocalDrive | ItemType.NetworkDrive | ItemType.PublicDrive;
 }
 
-export type OnItemOptionsClickHandler<T extends string = DefaultOptionId> = (
-    item: TreeItem<T>,
-    option: T,
-    drive: DriveTreeItem<T>,
+export type OnItemOptionsClickHandler = (
+    item: TreeItem,
+    option: DefaultOptionId,
+    drive: DriveTreeItem,
 ) => void;
 
-export interface DriveViewProps<T extends string = DefaultOptionId>
-    extends Omit<
+export interface DriveViewProps extends Omit<
         React.HTMLAttributes<HTMLDivElement>,
         'onDragEnd' | 'onDragStart'
     > {
     type: DriveType;
     name: string;
-    drives: DriveTreeItem<T>[];
-    defaultItemOptions?: ConnectTreeViewItemProps<T>['defaultOptions'];
-    onDropEvent?: (
-        item: DropItem<TreeItem<T>>,
-        target: TreeItem<T>,
-        event: DropEvent,
-        drive: DriveTreeItem<T>,
-    ) => void;
-    onItemClick?: (
-        event: React.MouseEvent<HTMLDivElement, MouseEvent>,
-        item: TreeItem<T>,
-        drive: DriveTreeItem<T>,
-    ) => void;
-    onSubmitInput?: (item: TreeItem, drive: DriveTreeItem<T>) => void;
-    onCancelInput?: (item: TreeItem, drive: DriveTreeItem<T>) => void;
-    onItemOptionsClick?: OnItemOptionsClickHandler<T>;
+    defaultItemOptions?: ConnectTreeViewItemProps['defaultOptions'];
+    drivePath?: string;
+    onDropEvent?: ConnectTreeViewProps['onDropEvent'];
+    onItemClick?: ConnectTreeViewProps['onItemClick'];
+    onSubmitInput?: ConnectTreeViewProps['onSubmitInput'];
+    onCancelInput?: ConnectTreeViewProps['onCancelInput'];
+    onItemOptionsClick?: ConnectTreeViewProps['onItemOptionsClick'];
     disableHighlightStyles?: boolean;
+    onDropActivate?: ConnectTreeViewProps['onDropActivate'];
+    onDragStart?: ConnectTreeViewProps['onDragStart'];
+    onDragEnd?: ConnectTreeViewProps['onDragEnd'];
```

Comment:
removed drive info from this callback functions. now you can get the drive of any item using the hook `useGetDriveParent`